cms/djangoapps/contentstore/views/course.py

```diff
@@ -849,16 +850,18 @@ def _create_or_rerun_course(request):
     Returns the destination course_key and overriding fields for the new course.
     Raises DuplicateCourseError and InvalidKeyError
     """
-    if not auth.user_has_role(request.user, CourseCreatorRole()):
-        raise PermissionDenied()
-
     try:
         org = request.json.get('org')
         course = request.json.get('number', request.json.get('course'))
         display_name = request.json.get('display_name')
         # force the start date for reruns and allow us to override start via the client
         start = request.json.get('start', CourseFields.start.default)
         run = request.json.get('run')
+        is_course_creator = (auth.user_has_role(request.user, CourseCreatorRole()) or
+                             auth.user_has_role(request.user, OrgContentCreatorRole(org=org)))
+
+        if not is_course_creator:
+            raise PermissionDenied()
```

Comment:
Ack, looks like we check course creator access in two other places as well:
* `_get_course_creator_status` in this same file
* `get_library_creator_status` in library.py

Both of those locations look at the `CourseCreator` table directly instead of looking at the auth roles. You will need to update both locations those to include the new organization-specificity logic.

Reply:
> _get_course_creator_status in this same file

This function is mostly responsible to view the list of content and IMHO this is okay to keep as is. I don't see any issues in this function, do let me know your thought about it.

Reply:
> get_library_creator_status

This function also looks at a higher level to see if the user has the permission to create a library, I think we should filter the Org part while creating the library.

Reply:
>  _get_course_creator_status in this same file

> This function is mostly responsible to view the list of content and IMHO this is okay to keep as is. I don't see any issues in this function, do let me know your thought about it

Hm, I do think the function should be made organization-aware. It would be incongruent if `library.get_library_creator_status` function knew about organizations, but `course._get_course_creator_status` did not take organizations into account.

Plus, that should enable you to keep the implementation logic:
```
(
    auth.user_has_role(user, CourseCreatorRole()) or
    auth.user_has_role(user, OrgContentCreatorRole(org=org)
)
```
within a single utility function, instead of repeating it in multiple places.